internal/controlplane/handlers_repositories_test.go

```diff
@@ -17,223 +17,132 @@ package controlplane
-					Return(tt.want.Result.Repository.HookUuid, stubbedHook, nil)
-			}
-
-			mockCryptoEngine := mockcrypto.NewMockEngine(ctrl)
-			mockCryptoEngine.EXPECT().DecryptOAuthToken(gomock.Any()).Return(oauth2.Token{AccessToken: accessToken}, nil).AnyTimes()
-
-			s := &Server{
-				store:           mockStore,
-				cryptoEngine:    mockCryptoEngine,
-				restClientCache: clientCache,
-				cfg:             &server.Config{},
-				evt:             &stubEventer,
-				webhookManager:  stubWebhookManager,
-			}
 			ctx := engine.WithEntityContext(context.Background(), &engine.EntityContext{
-				Provider: engine.Provider{Name: tt.req.Context.GetProvider()},
-				Project:  engine.Project{ID: projectUUID},
+				Provider: engine.Provider{Name: ghprovider.Github},
+				Project:  engine.Project{ID: projectID},
 			})
 
-			got, err := s.RegisterRepository(ctx, tt.req)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("Server.RegisterRepository() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
+			server := createServer(ctrl, scenario.RepoServiceSetup, scenario.ProviderFails)
 
-			if len(tt.events) != len(stubEventer.Sent) {
-				t.Fatalf("expected %d events, got %d", len(tt.events), len(stubEventer.Sent))
+			req := &pb.RegisterRepositoryRequest{
+				Repository: &pb.UpstreamRepositoryRef{
+					Owner: scenario.RepoOwner,
+					Name:  scenario.RepoName,
+				},
 			}
-			for i := range tt.events {
-				got := stubEventer.Sent[i]
-				want := &tt.events[i]
-				if !reflect.DeepEqual(got.Metadata, want.Metadata) {
-					t.Errorf("event %d.Metadata = %+v, want %+v", i, got.Metadata, want.Metadata)
-				}
-				var gotPayload, wantPayload EventPayload
-				if err := json.Unmarshal(got.Payload, &gotPayload); err != nil {
-					t.Fatalf("failed to unmarshal event %d.Payload: %v", i, err)
-				}
-				if err := json.Unmarshal(want.Payload, &wantPayload); err != nil {
-					t.Fatalf("failed to unmarshal event %d.Payload: %v", i, err)
+			res, err := server.RegisterRepository(ctx, req)
+			if scenario.ExpectedError == "" {
+				expectation := &pb.RegisterRepositoryResponse{
+					Result: &pb.RegisterRepoResult{
+						Repository: creationResult,
+						Status: &pb.RegisterRepoResult_Status{
+							Success: true,
+						},
+					},
 				}
-				if !reflect.DeepEqual(gotPayload.Repository, wantPayload.Repository) {
-					t.Errorf("event %d.Payload = %q, want %q", i, string(got.Payload), string(want.Payload))
+				require.NoError(t, err)
+				require.Equal(t, res, expectation)
+			} else {
+				// due to the mix of error handling styles in this endpoint, we
+				// need to do some hackery here
+				var errMsg string
+				if err != nil {
+					errMsg = err.Error()
+				} else if err == nil && res.Result.Status.Success == false && res.Result.Status.Error != nil {
+					errMsg = *res.Result.Status.Error
+				} else {
+					t.Fatal("expected error, but no error was found")
 				}
```

Comment:
Can you document and check for returning `Result.Repository` here, along with a link to the fix to prevent the client crash?  I'd like to preserve the institutional memory that there's a client crash if we return the wrong contents here, and a test feels like the way to do it.  (This is why I had the proto response in the original tests.)

Reply:
This code was leftover and needed to be deleted.

In the event of an error, a non-200 response will be returned and the CLI will not attempt to parse the response - this is pre-existing behaviour of the CLI.